script.js

```diff
@@ -0,0 +1,206 @@
+const TODOS = unparsedTODOS
+  ? JSON.parse(unparsedTODOS).map((todo, idx) => ({
+      ...todo,
+      idx,
+      fromDate: new Date(todo.fromDate),
+      toDate: new Date(todo.toDate),
+    }))
+  : [];
+
+let nextIdx = TODOS.length; // idx값을 중복되지 않게 설정하도록 초기값 지정
+const ul = document.querySelector('.todoList'); // child node를 추가/제거 하기 위해 변수에 저장
+
+// in memory에 저장된 todo list를 view에 display
+for (const todo of TODOS) {
+  addNewTodoLi(todo);
+}
+
+// 추가 버튼 click event handler 등록
+const addBtn = document.querySelector('.addBtn');
+addBtn.addEventListener('click', handleClickAddBtn);
+
+// 오늘 이전의 date은 선택하지 못하도록 제한
+const utc = Date.now();
+const timeOff = new Date().getTimezoneOffset() * 60000;
+const today = new Date(utc - timeOff).toISOString().split('T')[0];
+const dateInputs = document
+  .querySelectorAll('.dateInput')
+  .forEach((dateInput) => {
+    dateInput.setAttribute('min', today);
+  });
+
+// 시작일 설정에 change event handler 등록
+const fromDate = document.querySelector('.from');
+fromDate.addEventListener('change', handleChangeFromDate);
+
+// ############################## core functions ##############################
+// view update
+function updateView() {
+  while (ul.hasChildNodes()) {
+    // 기존에 추가됐든 li 삭제
+    ul.removeChild(ul.firstChild);
+  }
+  for (const todo of TODOS) {
+    // 업데이트된 todo list 추가
+    addNewTodoLi(todo);
+  }
+}
+
+// ul에 list들을 삽입
+function addNewTodoLi(todo) {
+  // li에 필요한 element들 선언
+  const newLi = document.createElement('li');
+  const contentDiv = document.createElement('div');
+  const fromDateDiv = document.createElement('div');
+  const betweenDiv = document.createElement('div');
+  const toDateDiv = document.createElement('div');
+  const doneBtn = document.createElement('button');
+  const deleteBtn = document.createElement('button');
+
+  // elements의 내용 추가
+  contentDiv.innerHTML = todo.content;
+  contentDiv.className = 'content';
+  fromDateDiv.innerHTML = convertDate(todo.fromDate);
+  fromDateDiv.className = 'fromDate';
+  betweenDiv.innerHTML = '-';
+  betweenDiv.className = 'betweenDate';
+  toDateDiv.innerHTML = convertDate(todo.toDate);
+  toDateDiv.className = 'toDate';
+  doneBtn.innerHTML = todo.isDone ? '↪' : '✓';
+  doneBtn.className = 'doneBtn';
+  deleteBtn.innerHTML = '-';
+  deleteBtn.className = 'deleteBtn';
+
+  // event listener 추가
+  doneBtn.addEventListener('click', handleClickDoneBtn);
+  deleteBtn.addEventListener('click', handleClickDeleteBtn);
+
+  // li에 child node들 추가
+  newLi.appendChild(contentDiv);
+  newLi.appendChild(fromDateDiv);
+  newLi.appendChild(betweenDiv);
+  newLi.appendChild(toDateDiv);
+  newLi.appendChild(doneBtn);
+  newLi.appendChild(deleteBtn);
+
+  newLi.id = todo.idx; // 정렬을 위한 id 추가
+  // 각종 className 추가
+  newLi.classList.add('todoLi'); 
+  if (todo.priority === 3) newLi.classList.add('high');
+  else if (todo.priority === 2) newLi.classList.add('mid');
+  else newLi.classList.add('low');
```

Comment:
혹시 이부분은 아직 작은 기능 구현이라 굳이 필요는 없을 것 같은데 이렇게 관리해도 괜찮을 것 같습니다!
```suggestion
const priorityClassMap = {
  3: 'high',
  2: 'mid',
  1: 'low'
};
const priorityClass = priorityClassMap[todo.priority] || 'low'; 
newLi.classList.add(priorityClass);

```